### PR TITLE
Fix datepicker

### PR DESCRIPTION
**Pull request by Thiblef**

### CODE REVIEW

File: BackofficeBundle/Form/Type/DatePickerType.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+use Symfony\Component\Form\Extension\Core\Type\DateType;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+
+/**
+ * Class DatePickerType
+ */
+class DatePickerType extends DateType
+{
+
```
Comment (nicolasThal) [inline BackofficeBundle/Form/Type/DatePickerType.php]: empty line to suppress

File: BackofficeBundle/Form/Type/DatePickerType.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    /**
+     * @param FormView $view
```
Comment (nicolasThal) [inline BackofficeBundle/Form/Type/DatePickerType.php]: you need to align the parameters

File: BackofficeBundle/Resources/config/form.yml [hunk 3]
```diff
@@ -313,3 +313,7 @@ services:
             - %open_orchestra_api.document.api_client.class%
         tags:
             - { name: form.type, alias: api_client }
+    open_orchestra_backoffice.type.date_picker:
+        class: %open_orchestra_backoffice.type.date_picker.class%
+        tags:
+            - { name: form.type, alias: date_picker}
```
Comment (nicolasThal) [inline BackofficeBundle/Resources/config/form.yml]: eof

File: BackofficeBundle/Form/Type/DatePickerType.php [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        }
+    }
+
+    /**
+     * Convert Symfony date format into jQuery date format
+     *
+     * @param $dateformat
+     * @return string jQuery compatible date format
+     */
+    public function convertDateFormat($dateformat)
```
Comment (nicolasThal) [inline BackofficeBundle/Form/Type/DatePickerType.php]: As this method is used only in the current class, you should put the visibility as protected only

File: BackofficeBundle/Form/Type/DatePickerType.php [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            // remove special characters unless the format was explicitly specified
+            if (!is_string($options['format'])) {
+                // remove quoted strings first
+                $pattern = preg_replace('/\'[^\']+\'/', '', $pattern);
+
+                // remove remaining special chars
+                $pattern = preg_replace('/[^yMd]+/', '', $pattern);
+            }
+
+            // set right order with respect to locale (e.g.: de_DE=dd.MM.yy; en_US=M/d/yy)
```
Comment (nicolasThal) [inline BackofficeBundle/Form/Type/DatePickerType.php]: The comments should be replaced by a test on the finishView method

File: BackofficeBundle/Tests/Form/Type/DatePickerTypeTest.php [hunk 6]
```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace OpenOrchestra\BackofficeBundle\Tests\Form\Type;
+
+use OpenOrchestra\BackofficeBundle\Form\Type\DatePickerType;
+
+/**
+ * Class DatePickerTypeTest
+ */
+class DatePickerTypeTest extends \PHPUnit_Framework_TestCase
```
Comment (nicolasThal) [inline BackofficeBundle/Tests/Form/Type/DatePickerTypeTest.php]: You should emphasize your test to test the finish view method in his whole scope

File: BackofficeBundle/Form/Type/DatePickerType.php [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+        $view->vars['widget'] = $options['widget'];
+        // Change the input to a HTML5 date input if
+        //  * the widget is set to "single_text"
+        //  * the format matches the one expected by HTML5
+        //  * the html5 is set to true
+        if ($options['html5'] && 'single_text' === $options['widget'] && self::HTML5_FORMAT === $options['format']) {
+            $view->vars['type'] = 'date';
+        }
+
+        if($options['widget'] === 'single_text') {
```
Comment (itkg-alavieille) [inline BackofficeBundle/Form/Type/DatePickerType.php]: the datePicker work only with the single_text option so why no force this options. 
If an user can user an other he uses the DateType and no DatePickerType

File: BackofficeBundle/Form/Type/DatePickerType.php [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    /**
+     * Returns the name of this type.
+     *
+     * @return string The name of this type
+     */
+    public function getName()
+    {
+        return 'date_picker';
+    }
+
```
Comment (itkg-alavieille) [inline BackofficeBundle/Form/Type/DatePickerType.php]: remove this empty line

File: BackofficeBundle/Resources/config/field_type.yml [hunk 9]
```diff
@@ -111,9 +111,9 @@ parameters:
                     required: true
         date:
             label: open_orchestra_backoffice.form.field_type.custom_type.date
-            type: date
+            type: date_picker
```
Comment (itkg-alavieille) [inline BackofficeBundle/Resources/config/field_type.yml]: accord with my previous remark, i think it's better to add a new field type for only datePicker

File: BackofficeBundle/Resources/public/coffee/orchestraLib.coffee [hunk 10]
```diff
@@ -135,6 +135,16 @@ activateOrchestraNodeChoice = (element) ->
         term.text.replace regExp, ''
     )
 
+#DATEPICKER ENABLED
+activateDatePicker = (element) ->
+  dataDateFormat = element.data('dateformat');
+  console.log dataDateFormat
```
Comment (itkg-alavieille) [inline BackofficeBundle/Resources/public/coffee/orchestraLib.coffee]: remove console.log

File: BackofficeBundle/Form/Type/DatePickerType.php [hunk 11]
```diff
@@ -0,0 +1,88 @@
+<?php
+
```
Comment (itkg-ngilain) [inline BackofficeBundle/Form/Type/DatePickerType.php]: empty line

File: BackofficeBundle/Form/Type/DatePickerType.php [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+     * @param FormInterface $form
+     * @param array $options
+     */
+    public function finishView(FormView $view, FormInterface $form, array $options)
+    {
+        $view->vars['widget'] = $options['widget'];
+        // Change the input to a HTML5 date input if
+        //  * the widget is set to "single_text"
+        //  * the format matches the one expected by HTML5
+        //  * the html5 is set to true
```
Comment (itkg-ngilain) [inline BackofficeBundle/Form/Type/DatePickerType.php]: You should remove this comment

File: BackofficeBundle/Resources/config/field_type.yml (hunk 9 above)
Comment (itkg-ngilain) [inline BackofficeBundle/Resources/config/field_type.yml]: I think it should be better to have only one date component, with several options to choose the rendering

**Pull request closed without merging**